src/overlays/actors/ovl_En_Test7/z_en_test7.c

```diff
@@ -32,80 +46,943 @@ const ActorInit En_Test7_InitVars = {
+
+            func_80AF0C30(this->unk_15C, &sp34, 1);
+            this->unk_144 |= 8;
+        }
+    } else {
+        this->unk_144 |= 1;
+    }
+}
+
+void func_80AF1E44(EnTest7* this, GlobalContext* globalCtx) {
+    Vec3f sp34;
+    Camera* camera;
+    f32 rand;
+
+    func_80AF1B68(this, globalCtx);
+
+    if (Rand_ZeroOne() < 0.3f) {
+        camera = Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(globalCtx->playerActorCsIds[8]));
+        rand = Rand_ZeroOne();
+        sp34.x = ((camera->eye.x - this->actor.world.pos.x) * rand) + this->actor.world.pos.x;
+        sp34.y = ((camera->eye.y - this->actor.world.pos.y) * rand) + this->actor.world.pos.y;
+        sp34.z = ((camera->eye.z - this->actor.world.pos.z) * rand) + this->actor.world.pos.z;
+        func_80AF0C30(this->unk_15C, &sp34, 1);
+    }
+
+    Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
+    func_80AF0C30(this->unk_15C, &sp34, 1);
+    this->unk_18FC[1].y += 0x2EE0;
+}
+
+void func_80AF1F48(EnTest7* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 temp = this->unk_1E54 - 86;
+    f32 temp_f0 = temp / 10.0f;
+    Vec3f sp20;
+
+    this->unk_144 |= 0x10;
+
+    this->unk_148.unk_08 = (-0.15f * temp_f0) + 0.25f;
+    this->unk_148.unk_0C = (-0.3f * temp_f0) + 0.4f;
+    this->unk_148.unk_10 -= 0x2EE0;
+
+    this->unk_144 |= 4;
+
+    if (this->unk_1E54 >= 96) {
+        func_80AF082C(this, func_80AF2030);
+        this->unk_144 &= ~0x10;
+    }
+    Math_Vec3f_Copy(&sp20, &this->actor.world.pos);
+    func_80AF0C30(this->unk_15C, &sp20, 1);
+}
+
+void func_80AF2030(EnTest7* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 temp = this->unk_1E54 - 96;
+    f32 sp1C = 1.0f - (temp / 4.00000000000000001f);
+    Camera* camera;
+    f32 temp_f2;
+    f32 temp_f4;
+
+    this->unk_148.unk_08 = ((temp * -0.1f) / 4.00000000000000001f) + 0.1f;
+    this->unk_148.unk_0C = ((temp * 5.9f) / 4.00000000000000001f) + 0.1f;
+    this->unk_148.unk_10 -= 0x2EE0;
+    this->actor.world.pos.y += 100.0f;
+
+    camera = Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(globalCtx->playerActorCsIds[8]));
+    camera->player = NULL;
+
+    camera->eye.x = ((camera->eye.x - this->unk_1E60.x) * sp1C) + this->unk_1E60.x;
+    camera->eye.y = ((camera->eye.y - this->unk_1E60.y) * sp1C) + this->unk_1E60.y;
+    camera->eye.z = ((camera->eye.z - this->unk_1E60.z) * sp1C) + this->unk_1E60.z;
+    camera->fov = ((camera->fov - this->unk_1E78) * sp1C) + this->unk_1E78;
+
+    if (this->unk_1E54 >= 100) {
+        MREG(64) = 1;
+        MREG(65) = 255;
+        MREG(66) = 255;
+        MREG(67) = 255;
+        MREG(68) = 255;
+        globalCtx->unk_18844 = 0;
+        this->unk_144 &= ~4;
+        func_80AF082C(this, func_80AF21E8);
+        func_80165690();
+    }
+}
+
+void func_80AF21E8(EnTest7* this, GlobalContext* globalCtx) {
+    s32 sp2C = this->unk_1E54 - 100;
+    f32 sp1C;
+    Color_RGB8 sp24 = D_80AF3434;
+    Color_RGB8 sp20 = D_80AF3438;
```

Comment:
```suggestion
    Color_RGB8 sp24 = D_80AF3434;
    Color_RGB8 sp20 = D_80AF3438;
```
Looks to be declared on the stack